frontend/src/components/admin/UserList/UserModal.js

```diff
@@ -0,0 +1,191 @@
+import React, {useEffect, useState} from 'react';
+import {Dialog, DialogActions, DialogContent, DialogTitle, Grid, TextField, Typography} from '@material-ui/core';
+import GCButton from '../../common/GCButton';
+import {GCCheckbox, styles, useStyles} from '../util/GCAdminStyles';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+import GameChangerAPI from '../../api/gameChanger-service-api';
+
+const gameChangerAPI = new GameChangerAPI();
+
+/**
+ * 
+ * @class UserModal
+ */
+export default ({showCreateEditUserModal, setShowCreateEditUserModal, userData, getUserData}) => {
+	const [editUserData, setEditUserData] = useState({});
+    
+
+	const classes = useStyles();
+
+	useEffect(()=>{
+		setEditUserData(userData);
+	},[userData])
+
+	const closeUserModal = () => {
+		setEditUserData({});
+		setShowCreateEditUserModal(false);
+	}
+	/**
+     * 
+     * @param {*} UserToEdit
+     * @returns 
+     */
+    
+	const storeUserData = (userToEdit = null) => {
+		if (!userToEdit) {
+			userToEdit = editUserData;
+		}
+
+		gameChangerAPI.storeUserData(userToEdit).then(data => {
+			if (data.status === 200) {
+				closeUserModal();
+				getUserData();
+			}
+		});
+	}
+
+	const handleCheck = (event) => {
+		setEditUserData({ ...editUserData, [event.target.name]: event.target.checked })
+	};
+
+	const handleTextChange = (event, key) => {
+		const tmpData = { ...editUserData };
+		tmpData[key] = event.target.value;
+		setEditUserData(tmpData);
+	};
```

Comment:
Should use React.useCallback for this to leverage a memoized callback to improve performance. See https://reactjs.org/docs/hooks-reference.html#usecallback

Reply:
Good call! I'll look into that. Thank you!

Reply:
@JLyons1985 is @jram930 even allowed to comment here?

Reply:
We will allow it, you on the other hand not so sure..... :)